configure.ac

```diff
@@ -28,9 +28,49 @@ AC_CONFIG_MACRO_DIR([m4/autoconf])
 
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 
-# To have empty CFLAGS instead of undefined and '-g -O2' by default
 CFLAGS=$CFLAGS
-CFLAGS="-Os -Wall -Wextra -Werror $CFLAGS"
+# To have empty CFLAGS instead of undefined and '-g -O2' by default
+
+### HOST FOR CROSS-COMPILING
+target_os=`uname -s`
+
+AC_MSG_CHECKING([target os])
+
+case "$target_os" in
+	MINGW* | cygwin* | msys* | mingw* | cegcc*)
+		CFLAGS="-g -g3 -Os -Wall -Wextra -std=c99 -Wformat $CFLAGS"
+		LDFLAGS="$LDFLAGS -Wl,--no-undefined"
+		target_os=mingw
+    ;;
+	ANDROID*)
+		target_os=android
+		CFLAGS="-Os -fPIE  -Wall -Wextra  $CFLAGS"
+		LDFLAGS="$LDFLAGS -pie"
+    ;;
+	    *)
+		target_os=linux
+		CFLAGS="-Os -Wall -Wextra -Werror $CFLAGS"
+AC_MSG_RESULT([$target_os])
```

Comment:
the target_os detection needs some work or maybe autotools has a variable kicking around. This only works on my local cross compiler toolchain as my unames are set by me...